src/libraries/System.Net.Sockets/src/System/Net/Sockets/NetworkStream.cs

```diff
@@ -337,6 +337,18 @@ public void Close(int timeout)
             Dispose();
         }
 
+        public void Close(TimeSpan timeout) => Close(ToTimeoutSeconds(timeout));
+
+        private static int ToTimeoutSeconds(TimeSpan timeout)
+        {
+            long totalSeconds = (long)timeout.TotalSeconds;
```

Comment:
I think the Socket.Close docs are wrong and thus this is, too.  The docs say it accepts a timeout in seconds, but on both Windows and Unix it's passed through raw to setsockopt, which expects milliseconds.

Reply:
> Is that true on .NET Framework also?

https://referencesource.microsoft.com/#System/net/System/Net/Sockets/Socket.cs,1152
https://referencesource.microsoft.com/#System/net/System/Net/Sockets/Socket.cs,6297
https://referencesource.microsoft.com/#System/net/System/Net/Sockets/Socket.cs,6331

Reply:
Oh good, that's easy then -- we just fix this line and update the docs.

Reply:
@danmoseley Accordingly, we should then modify ToTimeoutSeconds or?
Otherwise we have only a seconds precision?

Should I create a follow-up PR or how do we proceed?

Reply:
We should replace ToTimeoutSeconds here with ToTimeoutMilliseconds, since that's what the 'int' overload actually does, apparently. 

Then a follow up to fix https://github.com/dotnet/dotnet-api-docs/blob/main/xml/System.Net.Sockets/Socket.xml would be welcome.

Reply:
@danmoseley 
I will create a follow up (whether I still manage today in terms of time I can not currently judge).

Reply:
https://github.com/dotnet/dotnet-api-docs/pull/7888